component/src/main/java/com/synopsys/integration/alert/component/authentication/security/saml/SAMLContext.java

```diff
@@ -69,6 +70,24 @@ public boolean isSAMLEnabled() {
         return false;
     }
 
+    public void disableSAML() {
+        try {
+            ConfigurationModel configurationModel = getCurrentConfiguration();
+            Map<String, ConfigurationFieldModel> fields = configurationModel.getCopyOfKeyToFieldMap();
+            ConfigurationFieldModel enabledField = fields.get(AuthenticationDescriptor.KEY_SAML_ENABLED);
+            if (null != enabledField) {
+                enabledField.setFieldValue(String.valueOf(false));
+            } else {
+                enabledField = ConfigurationFieldModel.create(AuthenticationDescriptor.KEY_SAML_ENABLED);
+                enabledField.setFieldValue(String.valueOf(false));
+                fields.put(AuthenticationDescriptor.KEY_SAML_ENABLED, enabledField);
+            }
```

Comment:
This could be simplified to:
```suggestion
            if (null == enabledField) {
                enabledField = ConfigurationFieldModel.create(AuthenticationDescriptor.KEY_SAML_ENABLED);
                fields.put(AuthenticationDescriptor.KEY_SAML_ENABLED, enabledField);
            }
            enabledField.setFieldValue(String.valueOf(false));
```
If it feels weird to modify the reference after the `fields.put(...)` call, you could call `fields.put(...)` after the last line without any significant performance impact.

Reply:
That's not a problem.  I'll implement the change.